components/ConvertForm/ConvertForm.js

```diff
@@ -0,0 +1,374 @@
+import React, { useState, useCallback, useMemo } from 'react'
+import { OverlayTrigger, Tooltip } from 'react-bootstrap'
+import styled from 'styled-components'
+import AmountInput from 'components/AmountInput/AmountInput'
+import Anchor from 'components/Anchor/Anchor'
+import ConvertSteps from 'components/ConvertSteps/ConvertSteps'
+import LegalScreen from 'components/ConvertSteps/Legal'
+import NavBar from 'components/NavBar/NavBar'
+import Balance from 'components/SplitScreen/Balance'
+import SplitScreen from 'components/SplitScreen/SplitScreen'
+import { useWeb3Connect } from 'lib/web3-connect'
+import { useTokenBalance } from 'lib/web3-contracts'
+import { formatUnits } from 'lib/web3-utils'
+import { useConvertInputs } from './useConvertInputs'
+import { useOpenOrder, useClaimOrder, useApprove } from 'lib/web3-contracts'
+
+import question from './assets/question.svg'
+
+const options = ['ANT', 'ANJ']
+
+const CONVERTER_STATUSES = {
+  FORM: Symbol('STATE_FORM'),
+  LEGAL: Symbol('STATE_LEGAL'),
+  STEPPER: Symbol('STATE_STEPPER'),
+}
+
+function ConvertForm() {
+  const [formStatus, setFormStatus] = useState(CONVERTER_STATUSES.FORM)
+  const [selectedOption, setSelectedOption] = useState(1)
+  const [inverted, setInverted] = useState(true)
+  const toAnj = useMemo(() => !inverted, [inverted])
+  const {
+    amountSource,
+    amountRecipient,
+    bindOtherInput,
+    bondingPriceLoading,
+    handleManualInputChange,
+    inputValueRecipient,
+    inputValueSource,
+    resetInputs,
+  } = useConvertInputs(options[selectedOption], toAnj)
+  const tokenBalance = useTokenBalance(options[selectedOption])
+
+  const { account } = useWeb3Connect()
+
+  const openOrder = useOpenOrder()
+  const claimOrder = useClaimOrder()
+  const changeAllowance = useApprove()
+  const [buyOrderHash, setBuyOrderHash] = useState()
+
+  const conversionSteps = useMemo(() => {
+    const steps = [
+      [
+        'Create buy order',
+        {
+          createTx: () => openOrder(amountSource, toAnj),
+          hashCreated: hash => {
+            setBuyOrderHash(hash)
+          },
+        },
+      ],
+      [
+        'Claim order',
+        {
+          createTx: () => claimOrder(buyOrderHash, toAnj),
+        },
+      ],
+    ]
+
+    if (toAnj) {
+      steps.unshift(
+        [
+          'Reset approval',
+          {
+            createTx: () => changeAllowance(0),
+          },
+        ],
+        [
+          'Raise approval',
+          {
+            createTx: () => changeAllowance(amountSource),
+          },
+        ]
+      )
+    }
```

Comment:
@sohkai – I took a look at this on Sunday and landed on an array of tuples like you suggested. One thing I've yet to do is write the logic for when to show the extra reset and approval steps.

Currently in this code any `toAnj` conversion will add the reset and approval steps as seen here, that works reliably but I should only be adding these given the conditions mentioned [in your comment] (https://github.com/aragon/convert.aragon.org/pull/23#discussion_r399807156).

I'm a little confused on what exactly this logic should look like, which variables I should be sourcing here and what “allowance” and “approval” are actually doing behind the scenes. Perhaps it’s one to have a quick chat about offline though?